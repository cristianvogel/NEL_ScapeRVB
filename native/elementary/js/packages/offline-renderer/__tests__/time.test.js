import OfflineRenderer from '../index';
import { el } from '@elemaudio/core';


test('time node', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 1,
    numOutputChannels: 1,
  });

  // Graph
  core.render(el.time());

  // Ten blocks of data
  let inps = [new Float32Array(512 * 10)];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Push another small block. We should see an incrementing output beginning
  // at 512 * 10 because of the blocks we've already pushed through
  inps = [new Float32Array(32)];
  outs = [new Float32Array(32)];

  core.process(inps, outs);
  expect(outs[0]).toMatchSnapshot();
});

test('setting time', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 0,
    numOutputChannels: 1,
  });

  // Graph
  core.render(el.time());

  // Ten blocks of data
  let inps = [];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Set current time, then push another small block
  outs = [new Float32Array(8)];

  core.setCurrentTime(50);
  core.process(inps, outs);
  expect(Array.from(outs[0])).toMatchObject([50, 51, 52, 53, 54, 55, 56, 57]);

  // Set current time, then push another small block
  outs = [new Float32Array(8)];

  core.setCurrentTimeMs(1000);
  core.process(inps, outs);
  expect(Array.from(outs[0])).toMatchObject([44100, 44101, 44102, 44103, 44104, 44105, 44106, 44107]);
});
