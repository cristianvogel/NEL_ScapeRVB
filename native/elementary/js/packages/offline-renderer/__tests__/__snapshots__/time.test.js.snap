// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time node 1`] = `
Float32Array [
  5120,
  5121,
  5122,
  5123,
  5124,
  5125,
  5126,
  5127,
  5128,
  5129,
  5130,
  5131,
  5132,
  5133,
  5134,
  5135,
  5136,
  5137,
  5138,
  5139,
  5140,
  5141,
  5142,
  5143,
  5144,
  5145,
  5146,
  5147,
  5148,
  5149,
  5150,
  5151,
]
`;
